server/executor/eventemitter.go

```diff
@@ -0,0 +1,38 @@
+package executor
+
+import (
+	"context"
+
+	"github.com/kubeshop/tracetest/server/model"
+)
+
+type EventEmitter interface {
+	Emit(ctx context.Context, event model.TestRunEvent) error
+}
+
+type publisher interface {
+	Publish(eventID string, message any)
+}
+
+type internalEventEmitter struct {
+	repository model.TestRunEventRepository
+	publisher  publisher
+}
+
+func NewEventEmitter(repository model.TestRunEventRepository, publisher publisher) EventEmitter {
+	return &internalEventEmitter{
+		repository: repository,
+		publisher:  publisher,
+	}
+}
+
+func (em *internalEventEmitter) Emit(ctx context.Context, event model.TestRunEvent) error {
```

Comment:
On the next PRs we can evolve this method to also manage/start trace spans with each phase that we designed.